### ISSUE TITLE
CodeIntel support for ES6 Classes

### ISSUE THREAD

**Issue by Naatan** [Komodo/KomodoEdit#2518]
Komodo currently does not support ES6 classes for codeintel. Whilst classes still aren't supported in most browsers they are definitely gaining traction, as such we should add support for them soon.

**Comment by mitchell-as** [comment]
Perhaps #2158 supercedes this ticket @Naatan?

**Comment by Naatan** [comment]
Not exactly, ES6 is a bigger priority. ES7 is unlikely to be adopted widely any time soon considering ES6 isnt even entirely adopted.
